apps/dapp-console/app/components/Auth/RequireLogin.tsx

```diff
@@ -0,0 +1,51 @@
+import { useLogin, useLogout, usePrivy } from '@privy-io/react-auth'
+import { apiClient } from '@/app/helpers/apiClient'
+import { toast } from '@eth-optimism/ui-components'
+import { LONG_DURATION } from '@/app/constants/toast'
+import { captureError } from '@/app/helpers/errorReporting'
+import { useEffect, useState } from 'react'
+
+export const RequireLogin = ({ children }: { children: React.ReactNode }) => {
+  const { authenticated } = usePrivy()
```

Comment:
we should also check to make sure `ready` is `true` before we attempt to verify if the user is authenticated yet or not

Reply:
good catch, updated!